kythe/go/localrun/cmd/localrun/main.go

```diff
@@ -0,0 +1,205 @@
+)
+
+func init() {
+	flag.Var(&languages, "language", "Case insensitive list of languages that should be extracted and indexed")
+}
+
+var (
+	hostname        = flag.String("hostname", "localhost", "The host to start the http server on once everything is finished")
+	port            = flag.Int("port", 8080, "The port to start the http server on once everything is finished")
+	workingDir      = flag.String("working_dir", "", "The directory for all bazel oprations to begin relative to")
+	kytheRelease    = flag.String("kythe_release", "/opt/kythe", "The directory that holds a Kythe release. Releases can be downloaded from https://github.com/kythe/kythe/releases")
+	publicResources = flag.String("public_resources", "", "Path to the public resources to serve in the webserver (default: $kythe_release/resources/public)")
+	outputDir       = flag.String("output_dir", "", "The directory to create intermediate artifacts in")
+	indexingTimeout = flag.Duration("indexing_timeout", 300*time.Second, "How long to wait before indexing a compilation unit times out")
+	cacheSize       = datasize.Flag("cache_size", "3gb", "How much ram to dedicate to handling")
+)
+
+var errNotEnoughArgs = errors.New("not enough arguments")
+
+func getTargets(args []string) ([]string, error) {
+	if len(args) < 1 {
+		return nil, fmt.Errorf("You provided %d arguments but expected 2: %w", len(args), errNotEnoughArgs)
+	}
+	return args, nil
+}
+
+// usage prints out usage information for the program to stderr.
+func usage() {
+	fmt.Fprintf(os.Stderr, `localrun
+
+localrun is a small binary that can be used to build, extract, index, and serve any bazel repo.
+
+Usage:
+
+localrun [--language=a,b,c] [targets+]
+
+Flag arguments:
+
+`)
+	flag.Usage()
+
+	fmt.Fprintf(os.Stderr, `
+Positional arguments:
+
+  [targets+]: A list of one or more Bazel build targets that should be included in processing.
+
+Sample invocations:
+
+"bazel build" and generate cross references for all languages for everything:
+
+	localrun //...
+
+"bazel build" a portion of the repository and generate cross references for all languages:
+
+	localrun //just/this/path/...
+
+"bazel build" a portion of the repository (including non-go code) and generate cross references
+for just the go code:
+
+	localrun --language=go //just/this/path/...
+
+"bazel build" and generate cross references for just go and java code in the whole repo:
+
+	localrun --language=go,java //...
+
+Since this uses a standard bazel build invocation to build the targets, any targets that are
+included in the path will be included. If you would like to limit the build to just java_library
+targets you can employ bazel query to this end.
+
+Build only java_library targets and cross reference only java code:
+
+bazel query 'kind(java_library, //...)' | xargs localrun
+`)
+}
+
+// main entrypoint for the program.
+func main() {
+	flag.Parse()
+
+	*workingDir, _ = os.Getwd()
```

Comment:
Check error.

Also, the intention here seems to be that this flag overrides the working directory but it is just always ignored.  Either remove the flag or fix the override.

Reply:
Refactored it to be less bad